src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,119 @@
+    private static final String DUPLICATE_LOGIN = "DUPLICATE_LOGIN";
+    private static final int MIN_VALID_AGE = 18;
+    private static final int VALID_AGE = 60;
+    private static final int MAX_INVALID_AGE = 17;
+    private static final String VALID_PASSWORD = "123456";
+    private static final String INVALID_PASSWORD = "12345";
+    private User defaultUser;
+
+    @BeforeAll
+    public static void beforeAll() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        defaultUser = new User();
+        defaultUser.setLogin(DEFAULT_LOGIN);
+        defaultUser.setPassword(VALID_PASSWORD);
+        defaultUser.setAge(MIN_VALID_AGE);
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        service.register(defaultUser);
+        assertTrue(Storage.people.contains(defaultUser),
+                "Test failed! Expected storage people contains "
+                + defaultUser + " = true, but was false");
+    }
+
+    @Test
+    public void register_validUser_returnedEqualUser_ok() {
+        defaultUser.setLogin(THIRD_LOGIN);
+        User expectedUser = defaultUser;
+        User actualUser = service.register(defaultUser);
+        assertEquals(expectedUser, actualUser, "Test failed!"
+                + "Expected return value " + expectedUser
+                + ", but was " + actualUser);
+    }
+
+    @Test
+    public void register_validUser_sizeIncreased_ok() {
+        defaultUser.setLogin(FIRST_LOGIN);
+        int expectedSize = Storage.people.size() + 1;
+        service.register(defaultUser);
+        int actualSize = Storage.people.size();
+        assertEquals(expectedSize, Storage.people.size(), "Test failed!"
+                + " Size of storage after adding valid user should be "
+                + expectedSize + ", but was "
+                + actualSize
+        );
+    }
+
+    @Test
+    public void register_loginIsNull_notOk() {
+        defaultUser.setLogin(null);
+        assertThrows(UserValidationException.class, () -> service.register(defaultUser));
+    }
+
+    @Test
+    public void register_duplicateLogin_notOk() {
+        defaultUser.setLogin(DUPLICATE_LOGIN);
+        service.register(defaultUser);
+        assertThrows(UserExistsException.class, () -> service.register(defaultUser));
+    }
+
+    @Test
+    public void register_passwordIsNull_notOk() {
+        defaultUser.setLogin(SECOND_LOGIN);
+        defaultUser.setPassword(null);
+        assertThrows(UserValidationException.class, () -> service.register(defaultUser));
+    }
+
+    @Test
+    public void register_passwordLessThan6CharsLong_notOk() {
+        defaultUser.setLogin(FOURTH_LOGIN);
+        defaultUser.setPassword(INVALID_PASSWORD);
+        assertThrows(UserValidationException.class, () -> service.register(defaultUser));
+    }
+
+    @Test
+    public void register_ageLessThan18_notOk() {
+        defaultUser.setLogin(FIFTH_LOGIN);
```

Comment:
So, each time you add new test method, you have to add new login, it is not the most comfortable and scalable approach.
Let's better clean up the storage using AfterEach method.

Reply:
Thanks, I just through that I shouldn't use Storage directly, only Service